src/server/pkg/dbutil/dbutil.go

```diff
@@ -0,0 +1,53 @@
+package dbutil
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/pachyderm/pachyderm/src/client/pkg/require"
+)
+
+// set this to true if you want to keep the database around
+var devDontDropDatabase = false
+
+const (
+	DefaultPostgresHost = "127.0.0.1"
+	DefaultPostgresPort = 32228
+	TestPostgresUser    = "postgres"
+)
+
+// NewTestDB connects to postgres using the default settings, creates a database with a unique name
+// then calls cb with a sqlx.DB configured to use the newly created database.
+// After cb returns the database is dropped.
+func NewTestDB(t testing.TB) *sqlx.DB {
+	dsn := fmt.Sprintf("host=%s port=%d user=%s sslmode=disable", DefaultPostgresHost, DefaultPostgresPort, TestPostgresUser)
+	db := sqlx.MustOpen("postgres", dsn)
+	dbName := fmt.Sprintf("test_%d", time.Now().UnixNano())
+	db.MustExec("CREATE DATABASE " + dbName)
+	t.Log("database", dbName, "successfully created")
+	t.Cleanup(func() {
+		if !devDontDropDatabase {
+			db.MustExec("DROP DATABASE " + dbName)
+		}
+		require.NoError(t, db.Close())
+	})
+	db2 := sqlx.MustOpen("postgres", dsn+" dbname="+dbName)
```

Comment:
Why two handles?

Reply:
Each db is scoped to a postgres database.  The database is part of the DSN.  The first db is for setting up the database, then db2 is created scoped to that database, then db does the teardown.